app/controllers/admin/articles_controller.rb

```diff
@@ -1,7 +1,7 @@
 class Admin::ArticlesController < Admin::AdminController
   before_action :authorize
   before_action :set_article,              only: [:show, :edit, :update, :destroy]
-  before_action :set_contribution_options, only: [:new, :edit]
+  before_action :set_contribution_options, only: [:new, :edit, :update]
```

Comment:
Since `update` doesn't use the contributions data unless it fails to save (and render `:edit`), may as well not do the finds. You could just manually call `set_contribution_options` down on line 47 before `render :edit`.